### PR TITLE
Makes building with plasma glass a thing.

### LINKED ISSUE #30760: Can't place reinforced plasma glass on grilles.
Self-explanatory title, haven't tested with regular plasma glass though.

### PR DESCRIPTION
**Pull request by ShizCalev**
Closes #30760

Adds a helper, is_glass_sheet, which returns if the sheet you're using is... glass.

Makes it so that you can actually build plasma windows on grills, as well as being able to make fireproofed airlocks with plasma glass sheets.

### CODE REVIEW

**Comment by MrROBUST** [comment]
Will something like `is_glass_sheet` look better?

Can DM fill glass_sheet_types list via constructors of /obj/item/stack/sheet/glass and subtypes?

**Comment by ShizCalev** [comment]
Was on the fence about doing is_glass_sheet since only one other helper uses underscores, but I agree that it does look better.

File: code/game/objects/structures/grille.dm [hunk 1]
```diff
@@ -143,7 +143,11 @@
 				for(var/obj/structure/window/WINDOW in loc) //Another window already installed on grille
 					return
 				var/obj/structure/window/WD
-				if(istype(W, /obj/item/stack/sheet/rglass))
+				if(istype(W, /obj/item/stack/sheet/plasmarglass))
+					WD = new/obj/structure/window/plasma/reinforced/fulltile(loc) //reinforced plasma window
+				else if(istype(W, /obj/item/stack/sheet/plasmaglass))
+					WD = new/obj/structure/window/plasma/fulltile(loc) //plasma window
```
Comment (Cyberboss) [inline code/game/objects/structures/grille.dm]: These locs should be drop_location()

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/grille.dm [hunk 2]
```diff
@@ -249,10 +253,10 @@
 /obj/structure/grille/ratvar/New()
 	..()
 	if(broken)
-		new /obj/effect/temp_visual/ratvar/grille/broken(get_turf(src))
+		new /obj/effect/temp_visual/ratvar/grille/broken(drop_location())
 	else
-		new /obj/effect/temp_visual/ratvar/grille(get_turf(src))
-		new /obj/effect/temp_visual/ratvar/beam/grille(get_turf(src))
+		new /obj/effect/temp_visual/ratvar/grille(drop_location())
+		new /obj/effect/temp_visual/ratvar/beam/grille(drop_location())
```
Comment (Cyberboss) [inline code/game/objects/structures/grille.dm]: effects like these should probably stay on the turf

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/grille.dm [hunk 3]
```diff
@@ -117,13 +117,13 @@
 		if(!shock(user, 90))
 			user.visible_message("<span class='notice'>[user] rebuilds the broken grille.</span>", \
 								 "<span class='notice'>You rebuild the broken grille.</span>")
-			new grille_type(src.loc)
+			new grille_type(drop_location())
```
Comment (Cyberboss) [inline code/game/objects/structures/grille.dm]: Given that this is a transformation, it should use loc

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Pull request merged**